File: 2444_count_subarrays_with_fixed_bounds.cpp
class Solution {
public:
    long long countSubarrays(vector<int>& v, int p, int q) {
        long long z = 0;
        int a = -1, b = -1, c = -1;
        for (int i = 0; i < v.size(); ++i) {
            if (v[i] < p || v[i] > q) c = i;
            if (v[i] == p) a = i;
            if (v[i] == q) b = i;
            z += max(0, min(a, b) - c);
        }
        return z;
    }
};

File: 2444_count_subarrays_with_fixed_bounds.py
class Solution(object):
    def countSubarrays(self, x, y, z):
        r, u, v, w = 0, -1, -1, -1
        for i in range(len(x)):
            if x[i] < y or x[i] > z: w = i
            if x[i] == y: u = i
            if x[i] == z: v = i
            r += max(0, min(u, v) - w)
        return r
